tests/TestCIT_KCI.py

```diff
@@ -0,0 +1,290 @@
+                            for polyd in [1, 2]:
+                                cit_CIT = cit.CIT(data, 'kci', kernelX=kernelname, kernelY=kernelname,
+                                                  kernelZ=kernelname, est_width=est_width, use_gp=use_gp, approx=approx,
+                                                  polyd=polyd, kwidthx=kwidth, kwidthy=kwidth, kwidthz=kwidth)
+
+                                pvalue01.append(round(cit_CIT(0, 1), 3))
+                                # X and X_prime are independent, pvalue01 should be expected larger than 0.01
+                                pvalue03.append(round(cit_CIT(0, 3), 3))
+                                # X and Z are dependent, pvalue03 should be expected smaller than 0.01
+                                pvalue032.append(round(cit_CIT(0, 3, {2}), 3))
+                                # X and Z are independent conditional on Y, pvalue032 should be expected larger than 0.01
+
+        pvalue01_truth = [0.17, 0.17, 0.187, 0.169, 0.17, 0.17, 0.139, 0.191, 0.17, 0.17, 0.163, 0.169, 0.17, 0.17,
+                          0.154, 0.179, 0.17, 0.17, 0.167, 0.167, 0.17, 0.17, 0.163, 0.169, 0.557, 0.557, 0.533, 0.552,
+                          0.557, 0.557, 0.524, 0.549, 0.557, 0.557, 0.546, 0.557, 0.557, 0.557, 0.568, 0.528, 0.557,
+                          0.557, 0.528, 0.554, 0.557, 0.557, 0.537, 0.537, 0.17, 0.17, 0.167, 0.153, 0.17, 0.17, 0.168,
+                          0.18, 0.095, 0.095, 0.092, 0.104, 0.095, 0.095, 0.093, 0.091, 0.117, 0.117, 0.122, 0.119,
+                          0.117, 0.117, 0.113, 0.106, 0.636, 0.953, 0.64, 0.955, 0.636, 0.953, 0.618, 0.956, 0.636,
+                          0.953, 0.629, 0.95, 0.636, 0.953, 0.643, 0.963, 0.636, 0.953, 0.626, 0.958, 0.636, 0.953,
+                          0.622, 0.951, 0.636, 0.953, 0.629, 0.946, 0.636, 0.953, 0.636, 0.962, 0.636, 0.953, 0.634,
+                          0.957, 0.636, 0.953, 0.626, 0.955, 0.636, 0.953, 0.637, 0.955, 0.636, 0.953, 0.626, 0.947,
+                          0.636, 0.953, 0.628, 0.957, 0.636, 0.953, 0.667, 0.961, 0.636, 0.953, 0.651, 0.951, 0.636,
+                          0.953, 0.626, 0.951, 0.636, 0.953, 0.655, 0.971, 0.636, 0.953, 0.644, 0.951, 0.636, 0.636,
+                          0.613, 0.609, 0.636, 0.636, 0.599, 0.641, 0.636, 0.636, 0.608, 0.644, 0.636, 0.636, 0.663,
+                          0.645, 0.636, 0.636, 0.645, 0.647, 0.636, 0.636, 0.626, 0.649, 0.636, 0.636, 0.628, 0.641,
+                          0.636, 0.636, 0.625, 0.615, 0.636, 0.636, 0.657, 0.658, 0.636, 0.636, 0.649, 0.621, 0.636,
+                          0.636, 0.641, 0.633, 0.636, 0.636, 0.653, 0.613, 0.636, 0.636, 0.605, 0.65, 0.636, 0.636,
+                          0.647, 0.644, 0.636, 0.636, 0.623, 0.622, 0.636, 0.636, 0.638, 0.617, 0.636, 0.636, 0.654,
+                          0.644, 0.636, 0.636, 0.644, 0.617]
+        pvalue03_truth = [0.0] * (3 * 3 * 3 * 2 * 2 * 2)
```

Comment:
really? If we run the code, we can have all zeros?

Reply:
Yes.